harmonica/visualization/prism.py

```diff
@@ -0,0 +1,110 @@
+# Copyright (c) 2018 The Harmonica Developers.
+# Distributed under the terms of the BSD 3-Clause License.
+# SPDX-License-Identifier: BSD-3-Clause
+#
+# This code is part of the Fatiando a Terra project (https://www.fatiando.org)
+#
+"""
+Functions for visualizing prisms through pyvista
+"""
+import numpy as np
+
+try:
+    import pyvista
+except ImportError:
+    pyvista = None
+else:
+    import vtk
+
+
+def prisms_to_pyvista(prisms, properties=None):
+    """
+    Create a ``pyvista.UnstructuredGrid`` from a set of prisms
+
+    Parameters
+    ----------
+    prisms : list or 2d-array
+        List or 2d-array with the boundaries of the prisms.
+        Each row contains the boundaries of each prism in the following order:
+        ``west``, ``east``, ``south``, ``north``, ``bottom``, ``top``.
+    properties : dict or None (optional)
+        Dictionary with the physical properties of the prisms.
+        Each key should be a string and their value an array.
+        If None, no property will be added to the
+        :class:`pyvista.UnstructuredGrid`.
+        Default to None.
+
+    Returns
+    -------
+    pv_grid : :class:`pyvista.UnstructuredGrid`
+        :class:`pyvista.UnstructuredGrid` that represents the prisms with their
+        properties (if any).
+    """
+    # Check if pyvista are installed
+    if pyvista is None:
+        raise ValueError(
+            "Missing optional dependency 'pyvista' required for building pyvista grids."
+        )
+    # Get prisms and number of prisms
+    prisms = np.atleast_2d(prisms)
+    n_prisms = prisms.shape[0]
+    # Get the vertices of the prisms
+    vertices = np.vstack([_get_prism_vertices(prism) for prism in prisms])
+    # Generate the cells for the hexahedrons
+    cells = np.arange(n_prisms * 8).reshape([n_prisms, 8])
+    # Build the UnstructuredGrid
+    pv_grid = pyvista.UnstructuredGrid({vtk.VTK_HEXAHEDRON: cells}, vertices)
+    # Add properties to the grid
+    if properties is not None:
+        for name, prop in properties.items():
+            # Assign the property to the cell_data
+            pv_grid.cell_data[name] = np.atleast_1d(prop).ravel()
```

Comment:
Just a drive by note: you may want to double check that the array ordering is consistent here. Most of the time folks want `.ravel(order='F')` instead of the default `C` ordering. However since you are generating the cells for the UGrid explicitly, everything likely lines up

Reply:
I like how you are bringing a different point of view into it. I decided to use the same approach we are using on the prism forward modelling (if the density array is 2d, we ravel it), so both functions could take the same inputs and produce objects with the same logic.
https://github.com/fatiando/harmonica/blob/c1c2add5b0a40fd0104d5859b2d0048696f1954a/harmonica/forward/prism.py#L124

But you're right we are making an assumption here, and if any user is assuming the opposite, they wouldn't get a hint that something might be wrong with their results.
Maybe this is something we need to change project-wide. For now I think I can remove the raveling and ask for 1d properties arrays only. The `to_pyvista` method for the prism layer accessor would need to ravel it by itself, but in there we already have information about the correspondence between prism and its properties.

@leouieda  what do you think?

Reply:
I think this function should require 1D arrays since the prisms already have to be an unstructured 1D "list" of prisms anyway. Let the user ravel if needed.

Reply:
FWIW, +1 for leaving the reveling to users - always best to let users have control and be able to easily choose F vs C

Reply:
Thanks for the feedback. Done! 👍🏼